traceloader.py

```diff
@@ -0,0 +1,137 @@
+                mean: mean of bandwidth
+                var: variance of bandwidth
+                length: how much points generated
+                interval: how much points in one milleseconds
+
+        Returns:
+            dict, dict: traces, information of traces
+        '''
+        trace_dict = {}
+        trace_info = {}
+        seq = 0
+        for configs in configs_list:
+            seq = seq + 1
+            last_time = 0
+            timestamp = 0
+            transfer = 0
+            trace = ''
+            trace_name = ''
+            trace_desc = ''
+            raw_trace_data = np.array([])
+            for mean, var, length, interval in configs:
+                if trace_name == '':
+                    trace_name = f'{mean}-{seq}'
+                trace_desc = trace_desc + f'<{mean}-{var}-{length}-{interval}>'
+                bw = gauss(mean, var, length)
+                raw_trace_data = np.append(raw_trace_data, bw)
+                for b in bw: 
+                    transfer = transfer + b * 125 / interval
+                    last_time = last_time + 1
+                    if last_time >= interval:
+                        last_time = 0
+                        timestamp = timestamp + 1
+                        while transfer >= 1500:
+                            transfer = max(0, transfer - 1500)
+                            trace = trace + str(timestamp) + '\n'
+            trace_dict[trace_name] = trace
+            trace_info[trace_name] = {
+                'desc': trace_desc,
+                'length': timestamp,
+                'raw_data': raw_trace_data
+            }
+        return trace_dict, trace_info
+
+
+    def _link_trace_multibw_generator(self):
+        '''
+        Generate mahimahi traces that can have different bandwidth in different sections.
+
+        Args:
+            configs_list (list): each element is a list containing a bunch of lists. Every sub list has four elements: mean, var, length, interval
+                mean: mean of bandwidth
+                var: variance of bandwidth
+                length: how much points generated
+                interval: how much points in one milleseconds
+                e.g. [ [ [ 12, 2, 130000, 1], ], [ [ 48, 8, 130000, 1], [ 24, 6, 130000, 1], ],]
+            trace_path (str): folder to save traces
+
+        Returns:
+            dict: information of traces
+        '''
+        tools.clear_folder(self._trace_folder)
+        link_trace, trace_info = self._generate_mmlink_multibw(self._trace_list)
+        for trace in link_trace:
+            with open(os.path.join(self._trace_folder, trace), 'w', encoding='utf-8') as f:
+                f.write(link_trace[trace])
+        
+        if self._raw_save:
+            for info in trace_info:
+                raw_data = trace_info[info].pop('raw_data')
```

Comment:
Do you know what pop() means? It will delete the field of dict. So if you do like this, we will have different trace_info stored inside our class. That's not compatible.